pkg/events/definitions.go

```diff
@@ -39,6 +39,7 @@ func (d *definition) With(options ...Option) Builder {
 
 var defaultOptions = []Option{
 	WithVisibility(ttnpb.RIGHT_ALL),
+	WithUserAgentFromContext(),
```

Comment:
See discussion in #3068 for this

Reply:
Yes. @rvolosatovs made a good point there. The same applies here: it would be better to add this directly on events that are triggered in RPCs called by user agents.